bin/demeuk.py

```diff
@@ -118,9 +119,11 @@
 from unidecode import unidecode
 
 
-version = '3.9.6'
+version = '3.9.7'
 
-HEX_REGEX = re_compile(r'\$HEX\[([0-9a-f]+)\]')
+# Search from start to finish for the string $HEX[], with block of a-f0-9 with even number
+# of hex chars. The first match group is repeated.
+HEX_REGEX = re_compile(r"^\$HEX\[(([0-9a-f][0-9a-f])+)\]$", IGNORECASE)
```

Comment:
Lowercase `$hex[` or mixed-case `$HeX[` is now also a valid starting point, which is not tested:
```suggestion
HEX_REGEX = re_compile(r"^\$HEX\[(([0-9a-fA-F][0-9a-fA-F])+)\]$")
```

Optional: The un-used grouping could be removed, by using non-capturing group
```suggestion
HEX_REGEX = re_compile(r"^\$HEX\[((?:[0-9a-fA-F][0-9a-fA-F])+)\]$")
```

Optional: The regex could be simplified to make it the 'even number' claim more visible by using a repeated qualifier.
```suggestion
HEX_REGEX = re_compile(r"^\$HEX\[((?:[0-9a-fA-F]{2})+)\]$")
```

Reply:
Implemented some of the suggestions.